aviary/mission/flight_phase_builder.py

```diff
@@ -170,6 +175,37 @@ def build_phase(self, aviary_options: AviaryValues = None, phase_type=EquationsO
             rate_targets = ['dh_dr']
             rate2_targets = ['d2h_dr2']
 
+        # For multi-engine cases, we may have throttle allocation control.
+        if phase_type is EquationsOfMotion.HEIGHT_ENERGY and engine_count > 1:
+            allocation = user_options.get_val('throttle_allocation')
+
+            # Allocation should default to an even split so that we don't start
+            # with an allocation that might not produce enough thrust.
+            val = np.ones(engine_count - 1) * (1.0 / engine_count)
```

Comment:
Don't you want to assign a vale for each engine? You're only assigning a val for engine_count -1.

Reply:
The allocations all sum to 1.0, so you only need n-1 of them as variables.

Reply:
That should work well.